src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/migration/Cas1ValidateApAreaUserMapping.kt

```diff
@@ -0,0 +1,39 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.migration
+
+import org.slf4j.LoggerFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.client.ClientResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.client.CommunityApiClient
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.UserMappingService
+
+class Cas1ValidateApAreaUserMapping(
+  private val userRepository: UserRepository,
+  private val userMappingService: UserMappingService,
+  private val communityApiClient: CommunityApiClient,
+) : MigrationJob() {
+  private val log = LoggerFactory.getLogger(this::class.java)
+  override val shouldRunInTransaction = false
+
+  @SuppressWarnings("MagicNumber", "TooGenericExceptionCaught")
+  override fun process() {
+    userRepository.findAll().forEach { user ->
```

Comment:
I'll be interested to know how this behaves for CAS3 users. It doesn't make sense to record an AP Area for them, but I'm not sure we can distinguish between them from their db records once created. 

This is going to be a consideration in `UserService.getExistingUserOrCreate()` also -- as we'll want to set `User.ap_area_id` only for CAS1 users I think?

Reply:
As discussed, we'll only attempt to set/update the user.ap_area field if the call originated from the CAS1 UI (as we have no other way to identify if a given user will be using CAS1).